text/3-entrees/storage-api/vec-set.md

```diff
@@ -0,0 +1,145 @@
+# Using Vectors as Sets
+
+_[`pallets/vec-set`](https://github.com/substrate-developer-hub/recipes/tree/master/pallets/vec-set)_
+
+A [Set](<https://en.wikipedia.org/wiki/Set_(abstract_data_type)>) is an unordered data structure
+that stores entries without duplicates. Substrate's storage API does not provide a way to declare
+sets explicitly, but they can be implemented using either vectors or maps.
+
+This recipe demonstrates how to implement a storage set on top of a vector, and explores the performance of
+the implementation. When implementing a set in your own runtime, you should compare this technique
+to implementing a [`map-set`](./map-set.md).
+
+In this pallet we implement a set of `AccountId`s. We do not use the set for anything in this
+pallet; we simply maintain its membership. Using the set is demonstrated in the recipe on [pallet
+coupling](./pallet-couplin.md]. We provide dispatchable calls to add and remove members, ensuring
+that the number of members never exceeds a hard-coded maximum.
+
+```rust, ignore
+/// A maximum number of members. When membership reaches this number, no new members may join.
+pub const MAX_MEMBERS: usize = 16;
+```
+
+## Storage Item
+
+We will store the members of our set in a Rust
+[`Vec`](https://doc.rust-lang.org/std/vec/struct.Vec.html). A `Vec` is a collection of elements that
+is ordered and may contain duplicates. Because the `Vec` provides more functionality than our set
```

Comment:
`Vec`s are not ordered per se; did you mean to say "indexed"?

Reply:
Probably. What's the difference? What's a data structure that is "ordered" per se?

I want to communicate that you can iterate a `Vec` and get the items back in the same deterministic order every time. As opposed to something like a hash set.

Reply:
b-trees are sorted for instance, that's the go-to example I guess. Technically it depends a bit on the context: what does "ordered" mean? A `Vec` is "ordered" if the order you care about is insertion order.
I think "indexed" is the more technically correct term here for what you're trying to say. Maybe "A `Vec` is a collection of indexed elements in deterministic order…" or something along those lines?

Reply:
The rust docs say a `Vec` is "A contiguous growable array type". :)